### DISCUSSION TITLE
ProjectReference does not work in multi framework targeting project

### DISCUSSION THREAD

**Discussion by adydecki** [CommunityToolkit/Maui#351] [Q&A] [answered]
Hello team,
I am developing class library for Maui which depends on _net6-android/ios_ binding libraries. For this reason my csproj looks as follows:

> <Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <TargetFrameworks>net6.0;net6.0-android;net6.0-ios</TargetFrameworks>
        <UseMaui>true</UseMaui>
        <SingleProject>true</SingleProject>
    </PropertyGroup>
    <ItemGroup Condition="'$(TargetFramework)' == 'net6.0-android'">
        <ProjectReference Include="..\..\android\{binding_lib}net6.csproj" />
    </ItemGroup>
    <ItemGroup Condition="'$(TargetFramework)' == 'net6.0-ios'">
        <ProjectReference Include="..\..\ios\{binding_lib}net6.csproj" />
    </ItemGroup>
</Project>

However such configuration does not seem to work. I get following err msg:
`error NU1201: Project {binding_lib}net6 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0). 
Project {binding_lib}net6 supports: net6.0-ios15.2 (.NETCoreApp,Version=v6.0)`

I get different err msg based on the order of the frameworks listed in `TargetFrameworks` and when I switch the order in such a way the `net6.0-ios` is listed before `net6.0-android` the err msg is:
`error NU1201: Project {binding_lib}net6 is not compatible with net6.0-ios15.2 (.NETCoreApp,Version=v6.0). 
Project {binding_lib}net6 supports: net6.0-android31.0 (.NETCoreApp,Version=v6.0)`.

The error goes away when I only keep one `ProjectReference` and it does not matter either it is `net6.0-android` or `net6.0-ios`. Similar configuration works fine in my Xamarin.Forms class library when I reference monoandroid/xamarin.ios projects.

To me it seems like a bug, however first I would like to get some confirmation if this behavior is not expected and how to fix it.

I really appreciate any help you can provide.
Cheers,
Adrian.

**Comment by pictos** [comment, marked as answer]
@adydecki I would say that you should file this issue on the .NET MAUI repo. Anyway I'll ping @jonathanpeppers maybe he has some quick tip for this.

**Reply by adydecki** [reply]
Got it, thanks for your answer. I filed it under .NET MAUI repo at [https://github.com/dotnet/maui/discussions/5794](https://github.com/dotnet/maui/discussions/5794)

**Comment by CliffAgius** [comment]
Closed as answered
